From 057c75ebd31db4f560eacaaa06248c085bbd71d2 Mon Sep 17 00:00:00 2001
From: Xueling Li <xueling.li@amlogic.com>
Date: Thu, 28 May 2020 10:52:19 +0800
Subject: [PATCH] RDK: migrate code base to trunk [1/1]

PD#SWPL-24384

Problem:
  RDK current use predefined commit for most of AML recipes with a lot
  of additional patches
  Hard to move forward when AML codebase is update

Solution:
  Use RDK EXTERNAL_SRC mechanism for internal development
  manifest will check out all AML recipes

Verify:
  Local

Change-Id: Id3222ecd22012ca249b382eaa2da512459b661e1
Signed-off-by: Xueling Li <xueling.li@amlogic.com>
---
 liblog/include/cutils/list.h   | 30 ++++++++++++++++++++++--------
 liblog/include/cutils/logger.h |  8 ++++----
 logcat/Makefile                |  8 ++++++--
 logcat/logcat_test.c           | 11 +++++++++++
 4 files changed, 43 insertions(+), 14 deletions(-)
 create mode 100644 logcat/logcat_test.c

diff --git a/liblog/include/cutils/list.h b/liblog/include/cutils/list.h
index 83e702b..dfdc53b 100644
--- a/liblog/include/cutils/list.h
+++ b/liblog/include/cutils/list.h
@@ -19,7 +19,9 @@
 
 #include <stddef.h>
 
-#include <stddef.h>
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
 
 struct listnode
 {
@@ -32,20 +34,20 @@ struct listnode
 
 #define list_declare(name) \
     struct listnode name = { \
-        .next = &name, \
-        .prev = &name, \
+        .next = &(name), \
+        .prev = &(name), \
     }
 
 #define list_for_each(node, list) \
-    for (node = (list)->next; node != (list); node = node->next)
+    for ((node) = (list)->next; (node) != (list); (node) = (node)->next)
 
 #define list_for_each_reverse(node, list) \
-    for (node = (list)->prev; node != (list); node = node->prev)
+    for ((node) = (list)->prev; (node) != (list); (node) = (node)->prev)
 
 #define list_for_each_safe(node, n, list) \
-    for (node = (list)->next, n = node->next; \
-         node != (list); \
-         node = n, n = node->next)
+    for ((node) = (list)->next, (n) = (node)->next; \
+         (node) != (list); \
+         (node) = (n), (n) = (node)->next)
 
 static inline void list_init(struct listnode *node)
 {
@@ -61,6 +63,14 @@ static inline void list_add_tail(struct listnode *head, struct listnode *item)
     head->prev = item;
 }
 
+static inline void list_add_head(struct listnode *head, struct listnode *item)
+{
+    item->next = head->next;
+    item->prev = head;
+    head->next->prev = item;
+    head->next = item;
+}
+
 static inline void list_remove(struct listnode *item)
 {
     item->next->prev = item->prev;
@@ -71,4 +81,8 @@ static inline void list_remove(struct listnode *item)
 #define list_head(list) ((list)->next)
 #define list_tail(list) ((list)->prev)
 
+#ifdef __cplusplus
+};
+#endif /* __cplusplus */
+
 #endif
diff --git a/liblog/include/cutils/logger.h b/liblog/include/cutils/logger.h
index 97f2c56..58523fb 100644
--- a/liblog/include/cutils/logger.h
+++ b/liblog/include/cutils/logger.h
@@ -22,10 +22,10 @@ struct logger_entry {
     char        msg[0]; /* the entry's payload */
 };
 
-#define LOGGER_LOG_MAIN		"log/main"
-#define LOGGER_LOG_RADIO	"log/radio"
-#define LOGGER_LOG_EVENTS	"log/events"
-#define LOGGER_LOG_SYSTEM	"log/system"
+#define LOGGER_LOG_MAIN		"log_main"
+#define LOGGER_LOG_RADIO	"log_radio"
+#define LOGGER_LOG_EVENTS	"log_events"
+#define LOGGER_LOG_SYSTEM	"log_system"
 
 #define LOGGER_ENTRY_MAX_LEN		(4*1024)
 #define LOGGER_ENTRY_MAX_PAYLOAD	\
diff --git a/logcat/Makefile b/logcat/Makefile
index 19a47c5..ccdfd7e 100644
--- a/logcat/Makefile
+++ b/logcat/Makefile
@@ -7,15 +7,19 @@ OBJ = logcat.o
 CFLAGS = -Wall -Wextra -I $(LOGCAT_DIR)/include -DHAVE_IOCTL
 LDFLAGS = -L$(STAGING_DIR)/usr/lib -llog
 TARGET=logcat
+TEST := logcat_test
 
 # rules
-all: $(TARGET)
+all: $(TARGET) $(TEST)
 
 $(TARGET): $(OBJ)
+	$(CXX) $(LDFLAGS) $^ -o $@
+
+$(TEST): logcat_test.c
 	$(CC) $(LDFLAGS) $^ -o $@
 
 %.o: %.cpp
-	$(CC) -c $(CFLAGS) $^ -o $@
+	$(CXX) -c $(CFLAGS) $^ -o $@
 
 .PHONY: clean
 
diff --git a/logcat/logcat_test.c b/logcat/logcat_test.c
new file mode 100644
index 0000000..b48bcdb
--- /dev/null
+++ b/logcat/logcat_test.c
@@ -0,0 +1,11 @@
+#include <stdio.h>
+#include <cutils/log.h>
+#include <unistd.h>
+
+int main(int argc, char ** argv){
+    while (1){
+        sleep(1);
+        ALOGE("abert %s: test======>%d \n", __FUNCTION__, __LINE__);
+    }
+    return 0;
+}
-- 
2.17.1

